public/client.js

```diff
@@ -30,4 +30,18 @@ function removeProduct() {
   userRequest.open('post', '/removeProduct');
   userRequest.setRequestHeader("Content-Type", "application/json;charset=UTF-8")
   userRequest.send(JSON.stringify({'product':document.getElementById("removeProduct").value, 'quantity': document.getElementById("quantity").value}));
-}
\ No newline at end of file
+}
+
+
+function sortbyExpiry() {
+  var x = document.getElementById("expirysort").value;
+  window.location.replace("http://localhost:5000/?expirysort="+x);
```

Comment:
Same as the comment I left in #23, I forgot that this will always try to redirect to localhost even when this code is being run elsewhere. As a result, please modify to the following:
```
if (window.location.hostname === "localhost") {
    window.location.replace("http://localhost:5000/?expirysort="+x);
  } else {
    window.location.replace("https://whatscookinggoodlooking.herokuapp.com/?expirysort="+x);
  }
```